corehq/apps/locations/static/locations/js/location_types.js

```diff
@@ -321,6 +321,41 @@ hqDefine('locations/js/location_types.js', function(){
             };
         };
     }
+
+    $(function() {
```

Comment:
Definitely think we should be concerned about it.

In my head, I've been bucketing js files into "libraries" and "entry points" where the former don't have side effects and might be used by several pages, while the latter consist mostly of side effects (applying knockout bindings or setting up event handlers) and are typically used by only one page, or by a few closely-related pages. I'm not certain at this point that's the correct or only distinction to be making, might make sense to have a third "widget" bucket for files that define a widget or knockout model and then also initialize any instances of it.

When a js file is included on only one html template, I've been giving it the same name as that template, but otherwise I haven't been using any particular conventions. I think having  definitive conventions - and doing the necessary file renaming and code re-organization - will be part of the move to a module system, and will fit in with figuring out each page's entry point (I've been thinking we'll do a single entry point per page using something like [RequireJS's data-main attribute](http://requirejs.org/docs/api.html#data-main)) and also which scripts get included on which pages.

Happy to talk more, now or later, especially if you're thinking we should be making these decisions sooner rather than later.